lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -81,9 +81,11 @@ def problem_scores(self):
         ):
             block = self.course_data.structure[block_key]
             if getattr(block, 'has_score', False):
-                locations[block_key] = get_score(
+                problem_score = get_score(
                     submissions_scores={}, csm_scores={}, persisted_block=None, block=block,
                 )
+                if problem_score:
```

Comment:
I don't disagree with the end result because it makes the case we're handling more explicit, but isn't `problem_score` going to end up as an object, not an integer, and thus truthy any time it isn't `None`?